libs/blocks/mobile-app-banner/mobile-app-banner.js

```diff
@@ -48,8 +59,18 @@ function branchInit(key) {
       0
     );
     const privacyConsent = window.adobePrivacy?.hasUserProvidedConsent();
+    const isAndroid = navigator.userAgent.includes('Android');
+    
+    const cookieGrp = window.adobePrivacy?.activeCookieGroups();
+    const performanceCookieConsent = cookieGrp.includes('C0002');
+    const advertisingCookieConsent = cookieGrp.includes('C0004');
+
+    if ( performanceCookieConsent && advertisingCookieConsent && isAndroid ) {
+    branch.setBranchViewData({ data: { ecid: ecidVal }});
```

Comment:
Is it ok to set `ecid: null` ? I think you might want to check the value first

Reply:
If ecid is null, branch.setBranchViewData will automatically not set the ecid value. So no need to check if it is null.